Mathlib/Data/Multiset/Basic.lean

```diff
@@ -2682,6 +2682,42 @@ theorem disjoint_map_map {f : α → γ} {g : β → γ} {s : Multiset α} {t :
     Disjoint (s.map f) (t.map g) ↔ ∀ a ∈ s, ∀ b ∈ t, f a ≠ g b := by
   simp [Disjoint, @eq_comm _ (f _) (g _)]
 
+lemma notmem_singleton_cons_le [DecidableEq α] {a : α} {s : Multiset α} (h : a ∉ s)
+    {t : Multiset α} (ha : a ∈ t) (hs : s ≤ t) :
+    a ::ₘ s ≤ t := by
+  rw [← cons_erase ha, cons_le_cons_iff]
+  exact (cons_erase ha ▸ (@le_cons_of_not_mem _ _ (t.erase a) _ h)).mp hs
+
+lemma notmem_replicate_add_left_le [DecidableEq α] {a : α} {n : ℕ} {s : Multiset α} (h : a ∉ s)
```

Comment:
These two look like special cases of `add_le_of_disjoint` - I'd prefer to only give that one, and ensure that `s.Disjoint (replicate n a)` is easy to prove in the cases you care about.

Reply:
I thought I'd use `replicate` lemmas to prove `add_le_of_disjoint`. Maybe I'll just remove both `replicate` lemmas.